src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -586,12 +589,45 @@ private void ExecuteSyntaxTreeActions(AnalysisScope analysisScope, AnalysisState
                         cancellationToken.ThrowIfCancellationRequested();
 
                         // Execute actions for a given analyzer sequentially.
-                        AnalyzerExecutor.TryExecuteSyntaxTreeActions(syntaxTreeActions, analyzer, tree, analysisScope, analysisStateOpt, isGeneratedCode);
+                        AnalyzerExecutor.TryExecuteSyntaxTreeActions(syntaxTreeActions, analyzer, file, analysisScope, analysisStateOpt, isGeneratedCode);
 
                         processedAnalyzers?.Add(analyzer);
                     }
 
-                    analysisStateOpt?.MarkSyntaxAnalysisCompleteForUnprocessedAnalyzers(tree, analysisScope, processedAnalyzers);
+                    analysisStateOpt?.MarkSyntaxAnalysisCompleteForUnprocessedAnalyzers(file, analysisScope, processedAnalyzers);
+                }
+                finally
+                {
+                    processedAnalyzers?.Free();
+                }
+            }
+        }
+
+        private void ExecuteAdditionalFileActions(AnalysisScope analysisScope, AnalysisState analysisStateOpt, CancellationToken cancellationToken)
+        {
```

Comment:
In the equivalent method for source files we do some early bailout checks by looking to see if the `AnalysisScope` even cares about syntax files. Why don't we do equivalent checks here?